lib/folio_client/data_import.rb

```diff
@@ -19,18 +19,24 @@ def initialize(client)
     # @param job_profile_name [String] job profile name to use for import
     # @return [JobStatus] a job status instance to get information about the data import job
     def import(records:, job_profile_id:, job_profile_name:)
-      response_hash = client.post("/data-import/uploadDefinitions", {fileDefinitions: [{name: marc_filename}]})
+      response = client.connection.post("/data-import/uploadDefinitions", {fileDefinitions: [{name: marc_filename}]}.to_json, client.http_post_and_put_headers)
+
+      UnexpectedResponse.call(response) unless response.success?
+
+      response_hash = JSON.parse(response.body)
```

Comment:
The downside of the refactoring that brings parity between the Folio and Globus clients here is that we need to handle response parsing in more spots in the code. The upside is (arguably) easier reasoning about what's happening where.

Reply:
Most of the changes you'll see in this PR are of that variety.